Exiled.API/Features/Player.cs

```diff
@@ -1507,7 +1507,7 @@ public void Kill(string deathReason, string cassieAnnouncement = "")
             if (Side != Side.Scp && !string.IsNullOrEmpty(cassieAnnouncement))
                 Cassie.Message(cassieAnnouncement);
 
-            ReferenceHub.playerStats.KillPlayer(new CustomReasonDamageHandler(deathReason, float.MaxValue, cassieAnnouncement));
+            ReferenceHub.playerStats.DealDamage(new CustomReasonDamageHandler(deathReason, float.MaxValue, cassieAnnouncement));
```

Comment:
Right now this method will throw NRE, you should use `new CustomReasonDamageHandler(deathReason)`